# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/WSRunObject.h
# Commit: 4b43fd9bf671
# Full Hash: 4b43fd9bf67144f7f66e7f7bd8561807cae9de77
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2020-03-02 21:27:32
# Regressor Bug: 1618089
# File Overlap Count: 1
# Description:
#   Bug 1618089 - part 1: Redsign `EditorDOMPointBase` with template methods r=m_kato
#   
#   Some methods of `EditorDOMPointBase` assumes the container node type is
#   `nsINode`.  However, it's not good for reuse.  Therefore, this patch makes
#   most methods of them be template methods.
# ==============================================================================

diff -r b511a045d72e -r 4b43fd9bf671 editor/libeditor/WSRunObject.h
--- a/editor/libeditor/WSRunObject.h	Mon Mar 02 09:52:58 2020 +0000
+++ b/editor/libeditor/WSRunObject.h	Mon Mar 02 07:33:23 2020 +0000
@@ -259,11 +259,11 @@
    */
   MOZ_NEVER_INLINE_DEBUG EditorDOMPoint PointAfterContent() const {
     MOZ_ASSERT(mContent);
-    return mContent ? EditorDOMPoint::After(*mContent) : EditorDOMPoint();
+    return mContent ? EditorDOMPoint::After(mContent) : EditorDOMPoint();
   }
   MOZ_NEVER_INLINE_DEBUG EditorRawDOMPoint RawPointAfterContent() const {
     MOZ_ASSERT(mContent);
-    return mContent ? EditorRawDOMPoint::After(*mContent) : EditorRawDOMPoint();
+    return mContent ? EditorRawDOMPoint::After(mContent) : EditorRawDOMPoint();
   }
 
   /**
